modules/git/repo.go

```diff
@@ -55,6 +55,54 @@ func (repo *Repository) parsePrettyFormatLogToList(logs []byte) (*list.List, err
 	return l, nil
 }
 
+// RepoRefType type of repo reference
+type RepoRefType uint8
+
+const (
+	// RepoRefTypeInvalid invalid ref
+	RepoRefTypeInvalid RepoRefType = iota
+	// RepoRefTypeBranch branch
+	RepoRefTypeBranch
+	// RepoRefTypeTag tag
+	RepoRefTypeTag
+	// RepoRefTypeCommit commit
+	RepoRefTypeCommit
+	// RepoRefTypeBlob blob
+	RepoRefTypeBlob
+)
+
+// String converts RepoRefType to a the string representation
+func (t RepoRefType) String() string {
+	switch t {
+	case RepoRefTypeBlob:
+		return "blob"
+	case RepoRefTypeBranch:
+		return "branch"
+	case RepoRefTypeCommit:
+		return "commit"
+	case RepoRefTypeTag:
+		return "tag"
+	case RepoRefTypeInvalid:
+		return ""
+	default:
+		return ""
+	}
+}
+
+// GetRefType gets the type of the ref based on the string
+func (repo *Repository) GetRefType(ref string) RepoRefType {
```

Comment:
Could we return ObjectType on https://github.com/go-gitea/gitea/blob/master/modules/git/repo_object.go#L14 ?

Reply:
Looking into it. I actually realized I did not have this at all responding like GitHub, especially when it comes to being 4 different types of objects (dir, file, symlink, submodule) that could be returned, as well well as a list of entries. So this is being redone.

Reply:
Added ObjectBranch and now use those as a way to get the string of a ref. Also moved my GetRefType() into this file.